packages/element/src/serialize.js

```diff
@@ -618,6 +626,11 @@ export function renderAttributes( props ) {
  * @return {string} Style attribute value.
  */
 export function renderStyle( style ) {
+	// Only generate from object, e.g. tolerate string value.
```

Comment:
I was conflicted about this change, since it's largely aimed at the use-case being proposed here with `children` sourcing from DOM. In general, our custom serializer is much more tolerant to things which React would otherwise barf about. This has made the implementation slightly simpler in some regards too (though arguably not in this instance). Here, I saw no reason why `style` _must_ be an object, toward a goal where the fallback behavior is to always assign an attribute as given (and if it's given as a string, great: write it as-is). Other similar projects seem tolerant as well, e.g. [`preact-render-to-string`](https://github.com/developit/preact-render-to-string), which happily handles the string style. The main downside is a mixed expectation of what could be passed to `renderToString` vs. `render` alone.

If it's an issue, there's an alternative: We can detect the string from the `child` API's `fromDOM` and parse it back into an object.